ambari-server/src/main/java/org/apache/ambari/server/controller/internal/BlueprintConfigurationProcessor.java

```diff
@@ -1542,6 +1542,10 @@ public String updateForClusterCreate(String propertyName,
                 return origValue;
             }
 
+            if (topology.hasHadoopCompatibleFileSystem()) {
+              return origValue;
```

Comment:
I think this condition should also include a check that the component being processed is one from OneFS.  Otherwise it will fail to throw `IllegalArgumentException` for other components when it should.

To verify this, make the following change in `BlueprintConfigurationProcessorTest` and run it:

```diff
@@ -8244,7 +8244,11 @@ public class BlueprintConfigurationProcessorTest extends EasyMockSupport {

     replay(bp, topologyRequestMock);

-    ClusterTopology topology = new ClusterTopologyImpl(ambariContext, topologyRequestMock);
+    ClusterTopology topology = new ClusterTopologyImpl(ambariContext, topologyRequestMock) {
+      public boolean hasHadoopCompatibleFileSystem() {
+        return true;
+      }
+    };
     topology.setConfigRecommendationStrategy(ConfigRecommendationStrategy.NEVER_APPLY);

     return topology;
```

Result:

```
[ERROR]   BlueprintConfigurationProcessorTest.testDoUpdateForClusterCreate_SingleHostProperty__MissingComponent:2344 IllegalArgumentException should have been thrown
[ERROR]   BlueprintConfigurationProcessorTest.testDoUpdateForClusterCreate_SingleHostProperty__MultipleMatchingHostGroupsError:2386 IllegalArgumentException should have been thrown
```

Reply:
Nice catch.